git-changed-files.js

```diff
@@ -62,12 +62,36 @@ const isFileIgnored = (workingDirectory, file) => {
 /**
  * This lists the files that have changed when compared to `base` (a git ref),
  * limited to the files that are a descendent of `cwd`.
+ * It also respects '.gitattributes', filtering out files that have been marked
+ * as "binary" or "linguist-generated=true".
  */
 const gitChangedFiles = async (
     base /*:string*/,
     cwd /*:string*/,
 ) /*: Promise<Array<string>>*/ => {
     cwd = path.resolve(cwd);
+
+    // Github actions jobs can run the following steps to get a fully accurate
+    // changed files list. Otherwise, we fallback to a simple diff between the
+    // current and base branch, which might give false positives if the base
+    // is ahead of the current branch.
+    //
+    //   - name: Get All Changed Files
+    //     uses: jaredly/get-changed-files@absolute
+    //     id: changed
+    //     with:
+    //       format: 'json'
+    //       absolute: true
+    //
+    //   - uses: allenevans/set-env@v2.0.0
+    //     with:
+    //       ALL_CHANGED_FILES: '${{ steps.changed.outputs.added_modified }}'
+    //
+    if (process.env.ALL_CHANGED_FILES) {
+        const files = JSON.parse(process.env.ALL_CHANGED_FILES)
+        return files.filter(path => !isFileIgnored(cwd, path))
```

Comment:
You may have to manually run the build/lint process (I see that Prettier hasn't run here).

Reply:
@jeresig it is running, we just don't have it configured on this repo, so it's using defaults that don't add semicolons 🙃
I'll add a prettierrc

Reply:
Oh whoops! haha